File: CourierAmexClient/src/app/features/company/models/document-pay-type.model.ts
import { BaseEntity } from "@app/models";

export interface DocumentPayTypeModel extends BaseEntity<number> {
  companyId?: number;
  companyName?: string;
  name: string;
  payTypeId: number;
  payType?: string;
  currencyCode: number;
  currency?: string;
  bankId: number;
  brandId?: number;
  bankComission: number;
  vatWithholding: number;  
  incomeWithholding: number;
  moduleId?: string;
  templateId: number;
}


export const newDocumentPayType = {
  id: 0,
  companyId: undefined,
  name: '',
  payTypeId: 0,
  currencyCode: 0,
  bankId: 0,
  bankComission: 0,
  vatWithholding: 0,
  incomeWithholding: 0,
  moduleId: '',
  templateId: 0,
  status: 2
};

File: CourierAmexAPI/CourierAmex/Models/DocumentPayTypeModel.cs
namespace CourierAmex.Models
{ 
    public class DocumentPayTypeModel : BaseEntityModel<int>
    {
        public required string Name { get; set; }
        public required int PayTypeId { get; set; }
        public string? PayType { get; set; }
        public int CurrencyCode { get; set; }
        public string? Currency { get; set; }
        public int? BankId { get; set; }

        public int? BrandId { get; set; }

        public decimal BankComission { get; set; }

        public decimal VATWithholding { get; set; }

        public decimal IncomeWithholding { get; set; }

        public string? ModuleId { get; set; }

        public int TemplateId { get; set; }

        public required int CompanyId { get; set; }
        public string? CompanyName { get; set; }
    }
}

